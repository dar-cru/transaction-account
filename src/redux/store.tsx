import { Action, configureStore, Store } from "@reduxjs/toolkit";
import { ThunkAction } from "redux-thunk";
import rootReducer, { RootState } from "./rootReducer";

const store: Store = configureStore({ reducer: rootReducer });

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export default store;
