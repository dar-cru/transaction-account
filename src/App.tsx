import React from "react";
import { useSelector } from "react-redux";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import { RootState } from "./redux/rootReducer";
import NavigationBar from "./components/NavigationBar";
import AccountsList from "./views/AccountsList";
import CreateAccountForm from "./views/CreateAccountForm";
import LoginForm from "./views/LoginForm/LoginForm";
import TransactionsList from "./views/TransactionsList/TransactionsList.connected";

const App = () => {
  const { jwtToken } = useSelector((state: RootState) => state.authenticated);
  const isAuthenticated = !!jwtToken;
  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <>
            <NavigationBar isAuthenticated={isAuthenticated} />
            {isAuthenticated ? (
              <>
                <Route path={"/"} exact component={AccountsList} />
                <Route path={"/create"} exact component={CreateAccountForm} />
                <Route path={"/transactions"} component={TransactionsList} />
              </>
            ) : (
              <Route path={"/"} exact component={LoginForm} />
            )}
          </>
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
