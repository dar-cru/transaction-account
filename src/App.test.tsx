import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from './App';
import store from './redux/store';


describe('<App />', () => {
  test('renders the login form if there is no authentication', () => {
    const tree: RenderResult = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    const { container } = tree;
    expect(container).toMatchSnapshot();
  })
});
