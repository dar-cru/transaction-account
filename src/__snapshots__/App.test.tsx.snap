// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the login form if there is no authentication 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  background: #f2f2f2;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1rem;
  min-height: 60px;
}

.c1 {
  padding-left: 40px;
  font-size: 29px;
  color: darkolivegreen;
  font-family: sans-serif;
}

.c3 {
  min-height: 121px;
}

.c4 {
  width: 100%;
  padding: 12px 18px;
  margin: 8px 0;
  box-sizing: border-box;
  font-size: 16px;
}

.c5 {
  padding: 15px 48px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  cursor: pointer;
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  background-color: white;
  border: 2px solid #4CAF50;
  color: #4CAF50;
  box-shadow: none;
}

.c5:hover:not([disabled]) {
  background-color: #4CAF50;
  color: white;
  border: 2px solid #4CAF50;
  box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24),0 17px 50px 0 rgba(0,0,0,0.19);
}

.c5:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c2 {
  min-width: 200px;
  max-width: 660px;
  height: 348px;
  padding: 30px;
  border: 1px solid #ccc;
  margin: 30px auto;
}

<div>
  <div
    class="App"
  >
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        Coding Test
      </div>
    </nav>
    <form
      action="#"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <label
            for="username"
          >
            Username
             
          </label>
           
          <input
            class="c4"
            data-testid="input-username"
            name="username"
            value=""
          />
        </div>
        <div
          class="c3"
        >
          <label
            for="password"
          >
            Password
             
          </label>
           
          <input
            class="c4"
            data-testid="input-password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div>
          <button
            class="c5"
            data-testid="addButton"
            type="submit"
          >
            Login
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
